articles/ml_system_design/real_time_news_search_using_upstash_kafka_and_vector/upstash_ingest/upstash_ingest/models.py

```diff
@@ -0,0 +1,225 @@
+"""
+    Pydantic models definition used within the news data pipeline.
+    Implementation contains models for different NewsArticle formats:
+    - NewsDataIOModel: Model for NewsDataIO API response.
+    - NewsAPIModel: Model for NewsAPI response.
+    
+    Rest of the models are used for data processing and exchange within the pipeline:
+    - CommonDocument: Common representation of a news article.
+    - RefinedDocument: Refined version of a CommonDocument.
+    - ChunkedDocument: Chunked version of a RefinedDocument.
+    - EmbeddedDocument: Embedded version of a ChunkedDocument.
+     
+"""
+
+import datetime
+import hashlib
+import logging
+from typing import Any, Dict, List, Optional, Union
+from uuid import uuid4
+
+from dateutil import parser
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from pydantic import BaseModel, Field, field_validator
+from unstructured.staging.huggingface import chunk_by_attention_window
+
+from .cleaners import clean_text, normalize_whitespace, remove_html_tags
+from .embeddings import TextEmbedder
+
+logger = logging.getLogger(__name__)
+
+RECURSIVE_SPLITTER = RecursiveCharacterTextSplitter()
+
```

Comment:
I don't think it is necessary to add this as a global variable

Reply:
Moving it as part of a pydantic model, will mean it is instantiated every time a new document is created.
Since it requires no arguments, having it globally defined removes rendudancy.